apps/src/p5lab/spritelab/constants.js

```diff
@@ -100,6 +100,9 @@ export const exampleSprites = {
 
 // Big numbers in some blocks can cause performance issues. Combined with live-preview,
 // this results in hanging the tab and students unable to edit their blocks. We
-// guard against this by capping numbers where needed.
-export const MAX_NUM_SPRITES = 1000;
+// guard against this by capping number of sprites.
+// The user will receive a workspace alert at MAX_NUM_SPRITE_WARNING. But the actual cap
+// is MAX_NUM_SPRITES
+export const MAX_NUM_SPRITES = 1001;
+export const MAX_NUM_SPRITE_WARNING = MAX_NUM_SPRITES - 1;
```

Comment:
I'm concerned that the MAX_NUM_SPRITES and MAX_NUM_SPRITE_WARNING could lead to future off-by-one errors that are difficult to catch. A future code maintainer would need to be very careful that they're using the right variable in the right place. Since 1000 is a somewhat arbitrary limit (i.e. choosing 1010 or 995 would lead to no real difference in user experience), just using `MAX_NUM_SPRITES = 1000` and limiting the user if `spritesRequested > MAX_NUM_SPRITES` and displaying the warning if `spritesRequested === MAX_NUM_SPRITES` seems very reasonable to me. Then there's only one variable that needs to be used everywhere and it's a little more clear to the reader what is happening in each place.

I'm curious to hear others thoughts, however. This is how I would approach it, but I could see an argument in a different direction.

Reply:
Yes - I would also like to hear other folks's thoughts as well!
I renamed the MAX_NUM_SPRITE_WARNING to SPRITE_LIMIT_WARNING and made other updates in its use (created a 2nd function) and tests. More details in my reply to comment above. Thanks!

Reply:
I agree with @jmkulwik that it seems a bit odd to export two numbers, one of which is always one less than the other, when you really only need to export one of them, and the business logic elsewhere can handle the warning when you are exactly one before.  However, there is one argument for keeping them as separate, and that would be if we might want to adjust this to warn at a notably different threshold than the maximum.  If that's never going to happen, though, then it seems more robust to just export one magic number.

Reply:
Thanks @breville ! This makes and I'm definitely open to refactor.